src/incremental.rs

```diff
@@ -1,17 +1,44 @@
 // Copyright 2023 Paxos
 
 //! Logic for incremantal runs
+use crate::{
+    mutate::{Mutant, MutantHash},
+    options::Options,
+    output::{PositiveOutcome, OUTDIR_NAME, POSITIVE_OUTCOMES_FILE},
+};
+use anyhow::{anyhow, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use std::{collections::HashSet, fs};
 
-use crate::{mutate::Mutant, output::PositiveOutcomes};
-
-pub fn filter(mutants: Vec<Mutant>) -> (Option<PositiveOutcomes>, Vec<Mutant>) {
-    let last_positive_outcomes = read_last_positive_outcomes();
-    // TODO Exclude mutants whose hash is in the last positive outcomes
-    let mutants = mutants.into_iter().filter(|m| todo!());
-
-    (last_positive_outcomes, mutants.collect())
+pub fn filter_by_last_positive_outcomes(
+    mutants: Vec<Mutant>,
+    dir: &Utf8PathBuf,
+    options: &Options,
+) -> (Option<Vec<PositiveOutcome>>, Vec<Mutant>) {
+    let read_path: &Utf8Path = options.output_in_dir.as_ref().map_or(dir, |p| p.as_path());
+    // TODO: add logging here for error cases
+    let last_positive_outcomes = match read_last_positive_outcomes(read_path) {
+        Ok(outcomes) => Some(outcomes),
+        Err(_) => None,
+    };
```

Comment:
Right, because if the file is not there, it just means that it’s a first run, but otherwise we are in trouble?